src/ICeWall/src/ICeWall.tcl

```diff
@@ -1,9 +1,147 @@
+                                           [-rdl_spacing value] \
+                                           [-rdl_cover_file_name rdl_file_name]}
+
+proc set_bump_options {args} {
+  sta::parse_key_args "set_bump_options" args \
+    keys {-pitch -bump_pin_name -spacing_to_edge -cell_name -num_pads_per_tile -rdl_layer -rdl_width -rdl_spacing -rdl_cover_file_name}
+
+  ICeWall::set_bump_options {*}[array get keys]
+  if {[info exists keys(-rdl_cover_file_name)]} {
+    ICeWall::set_rdl_cover_file_name $keys(-rdl_cover_file_name)
+  }
+}
+
+sta::define_cmd_args "set_padring_options" {[-type (flipchip|wirebond)] \
+                                            [-power power_nets] \
+                                            [-ground ground_nets] \
+                                            [-core_area core_area] \
+                                            [-die_area die_area] \
+                                            [-offsets offsets] \
+                                            [-pad_inst_name pad_inst_name] \
+                                            [-pad_pin_name pad_pin_name] \
+                                            [-pin_layer pin_layer_name] \
+                                            [-connect_by_abutment signal_list]}
+
+proc set_padring_options {args} {
+  sta::parse_key_args "set_padring_options" args \
+    keys {-type -power -ground -core_area -die_area -offsets -pad_inst_name -pad_pin_name -pin_layer -connect_by_abutment}
+
+  if {[info exists keys(-type)]} {
+    ICeWall::set_type $keys(-type)
+  }
+
+  if {[info exists keys(-power)]} {
+    ICeWall::add_power_nets {*}$keys(-power)
+  }
+
+  if {[info exists keys(-ground)]} {
+    ICeWall::add_ground_nets {*}$keys(-ground)
+  }
+
+  if {[info exists keys(-core_area)]} {
+    ICeWall::set_core_area {*}$keys(-core_area)
+  }
+
+  if {[info exists keys(-die_area)]} {
+    ICeWall::set_die_area {*}$keys(-die_area)
+  }
+
+  if {[info exists keys(-offsets)]} {
+    ICeWall::set_offsets $keys(-offsets)
+  }
+
+  if {[info exists keys(-pad_inst_name)]} {
+    ICeWall::set_pad_inst_name $keys(-pad_inst_name)
+  }
+
+  if {[info exists keys(-pad_pin_name)]} {
+    ICeWall::set_pad_pin_name $keys(-pad_pin_name)
+  }
+
+  if {[info exists keys(-pin_layer)]} {
+    ICeWall::set_pin_layer $keys(-pin_layer)
+  }
+
+  if {[info exists keys(-connect_by_abutment)]} {
+    ICeWall::set_library_connect_by_abutment {*}$keys(-connect_by_abutment)
+  }
+}
+
+sta::define_cmd_args "add_libcell" {[-name name] \
+                                      [-type cell_type] \
+                                      [-cell_name cell_names_per_side] \
+                                      [-orient orientation_per_side] \
+                                      [-pad_pin_name pad_pin_name] \
+                                      [-break_signals signal_list] \
+                                      [-physical_only]}
+
+proc add_libcell {args} {
```

Comment:
If possible, you might want to consider renaming this command from add_libcell (which seem to imply you we a adding library cells) to something like: add_pad_libcell (making it clear that this command belongs to the padring)

Reply:
I will change it add_pad_libcell then - any other opionins on this?

Reply:
define_pad_cell
you aren't really adding it, you are defining it

Reply:
Yes - that sounds better.